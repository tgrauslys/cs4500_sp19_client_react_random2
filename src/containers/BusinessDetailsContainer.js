import React from 'react'

import BusinessDetails from "../components/BusinessDetails";

class BusinessDetailsContainer extends React.Component {
    constructor(props) {
        super(props)
        //this.businessService = this.props.service
        this.stateFieldNames = ['name',
            'yearFounded',
            'employees',
            'email',
            'street',
            'city',
            'state',
            'zip',
            'paymentMethods',
            'facebook',
            'instagram',
            'twitter']
        this.state = {
            business: {
                name: 'Example',
                yearFounded: '2000',
                employees: '20',
                email: 'an email',
                street: 'a street',
                city: 'a city',
                state: 'a state',
                zip: 'a zip',
                paymentMethods: 'some payment methods',
                facebook: 'a facebook link',
                instagram: 'an instagram link',
                twitter: 'a twitter link'
            }
        }
        this.handleChange = this.handleChange.bind(this)
    }

    /*    componentDidMount() {
            this.findAllQuotes()
        }
        findAllBusinesses = () =>
            this.businessService
                .findAllBusinesses()
                .then(quotes => {
                        this.setState({
                            quotes: quotes
                        })
                    }
                )

        createBusiness = () =>
            this.quoteService
                .createQuote(this.state.quote)
                .then(this.findAllBusinesses)
        deleteBusiness = id =>
            this.quoteService
                .deleteQuote(id)
                .then(this.findAllBusinesses)
        updateBusiness = () =>
            this.quoteService
                .updateBusiness(this.state.business)
                .then(this.findAllBusinesses())*/
    handleChange(e) {
        let change = {business: {}}

        this.stateFieldNames.forEach(fieldName => {
            if (fieldName === e.target.id){
                change["business"][e.target.id] = e.target.value
            } else {
                change["business"][fieldName] = this.state.business[fieldName]
            }
        })
        this.setState(change)
    }

    render() {
        return (
            <div>
                <BusinessDetails
                    // updateBusiness={this.updateBusiness}
                    // deleteBusiness={this.deleteBusiness}
                    // createBusiness={this.createBusiness}
                    // updateForm={this.updateForm}
                    handleChange={this.handleChange}
                    business={this.state.business}/>
            </div>
        )
    }
}

export default BusinessDetailsContainer