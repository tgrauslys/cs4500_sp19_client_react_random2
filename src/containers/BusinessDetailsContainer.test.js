import React from 'react'
import BusinessDetailsContainer from './BusinessDetailsContainer'
import TestRenderer from 'react-test-renderer';

test('BusinessDetailsContainer renders correctly', async () => {
    const testRenderer = TestRenderer.create(
        <BusinessDetailsContainer/>)
    let tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const testInstance   = testRenderer.root
    let name = testInstance.findByProps({id: 'name'})

    name.props.onChange({target: {id:"name", value: 'Different'}})
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    let year = testInstance.findByProps({id: 'yearFounded'})

    year.props.onChange({target: {id: 'yearFounded', value: '1'}})
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    let employees = testInstance.findByProps({id: 'employees'})

    employees.props.onChange({target: {id: 'employees', value: '321'}})
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

})