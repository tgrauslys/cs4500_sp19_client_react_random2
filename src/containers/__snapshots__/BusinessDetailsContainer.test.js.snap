// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessDetailsContainer renders correctly 1`] = `
<div>
  <div>
    <h2>
      Business
    </h2>
    Business Name
    <input
      id="name"
      onChange={[Function]}
      value="Example"
    />
    <br />
    Year Founded
    <input
      id="yearFounded"
      onChange={[Function]}
      value="2000"
    />
    <br />
    Number of Employees
    <input
      id="employees"
      onChange={[Function]}
      value="20"
    />
    <br />
    Email
    <input
      id="email"
      onChange={[Function]}
      value="an email"
    />
    <br />
    <h3>
      Business Address (optional)
    </h3>
    <br />
    Street
    <input
      id="street"
      onChange={[Function]}
      value="a street"
    />
    <br />
    City
    <input
      id="city"
      onChange={[Function]}
      value="a city"
    />
    <br />
    State
    <input
      id="state"
      onChange={[Function]}
      value="a state"
    />
    <br />
    Zip
    <input
      id="zip"
      onChange={[Function]}
      value="a zip"
    />
    <br />
    <h2>
      Payment Methods
    </h2>
    <input
      id="paymentMethods"
      onChange={[Function]}
      value="some payment methods"
    />
    <br />
    <h2>
      Social Media
    </h2>
    Facebook
    <input
      id="facebook"
      onChange={[Function]}
      value="a facebook link"
    />
    <br />
    Instagram
    <input
      id="instagram"
      onChange={[Function]}
      value="an instagram link"
    />
    <br />
    Twitter
    <input
      id="twitter"
      onChange={[Function]}
      value="a twitter link"
    />
    <br />
    <button>
      SAVE
    </button>
  </div>
</div>
`;

exports[`BusinessDetailsContainer renders correctly 2`] = `
<div>
  <div>
    <h2>
      Business
    </h2>
    Business Name
    <input
      id="name"
      onChange={[Function]}
      value="Different"
    />
    <br />
    Year Founded
    <input
      id="yearFounded"
      onChange={[Function]}
      value="2000"
    />
    <br />
    Number of Employees
    <input
      id="employees"
      onChange={[Function]}
      value="20"
    />
    <br />
    Email
    <input
      id="email"
      onChange={[Function]}
      value="an email"
    />
    <br />
    <h3>
      Business Address (optional)
    </h3>
    <br />
    Street
    <input
      id="street"
      onChange={[Function]}
      value="a street"
    />
    <br />
    City
    <input
      id="city"
      onChange={[Function]}
      value="a city"
    />
    <br />
    State
    <input
      id="state"
      onChange={[Function]}
      value="a state"
    />
    <br />
    Zip
    <input
      id="zip"
      onChange={[Function]}
      value="a zip"
    />
    <br />
    <h2>
      Payment Methods
    </h2>
    <input
      id="paymentMethods"
      onChange={[Function]}
      value="some payment methods"
    />
    <br />
    <h2>
      Social Media
    </h2>
    Facebook
    <input
      id="facebook"
      onChange={[Function]}
      value="a facebook link"
    />
    <br />
    Instagram
    <input
      id="instagram"
      onChange={[Function]}
      value="an instagram link"
    />
    <br />
    Twitter
    <input
      id="twitter"
      onChange={[Function]}
      value="a twitter link"
    />
    <br />
    <button>
      SAVE
    </button>
  </div>
</div>
`;

exports[`BusinessDetailsContainer renders correctly 3`] = `
<div>
  <div>
    <h2>
      Business
    </h2>
    Business Name
    <input
      id="name"
      onChange={[Function]}
      value="Different"
    />
    <br />
    Year Founded
    <input
      id="yearFounded"
      onChange={[Function]}
      value="1"
    />
    <br />
    Number of Employees
    <input
      id="employees"
      onChange={[Function]}
      value="20"
    />
    <br />
    Email
    <input
      id="email"
      onChange={[Function]}
      value="an email"
    />
    <br />
    <h3>
      Business Address (optional)
    </h3>
    <br />
    Street
    <input
      id="street"
      onChange={[Function]}
      value="a street"
    />
    <br />
    City
    <input
      id="city"
      onChange={[Function]}
      value="a city"
    />
    <br />
    State
    <input
      id="state"
      onChange={[Function]}
      value="a state"
    />
    <br />
    Zip
    <input
      id="zip"
      onChange={[Function]}
      value="a zip"
    />
    <br />
    <h2>
      Payment Methods
    </h2>
    <input
      id="paymentMethods"
      onChange={[Function]}
      value="some payment methods"
    />
    <br />
    <h2>
      Social Media
    </h2>
    Facebook
    <input
      id="facebook"
      onChange={[Function]}
      value="a facebook link"
    />
    <br />
    Instagram
    <input
      id="instagram"
      onChange={[Function]}
      value="an instagram link"
    />
    <br />
    Twitter
    <input
      id="twitter"
      onChange={[Function]}
      value="a twitter link"
    />
    <br />
    <button>
      SAVE
    </button>
  </div>
</div>
`;

exports[`BusinessDetailsContainer renders correctly 4`] = `
<div>
  <div>
    <h2>
      Business
    </h2>
    Business Name
    <input
      id="name"
      onChange={[Function]}
      value="Different"
    />
    <br />
    Year Founded
    <input
      id="yearFounded"
      onChange={[Function]}
      value="1"
    />
    <br />
    Number of Employees
    <input
      id="employees"
      onChange={[Function]}
      value="321"
    />
    <br />
    Email
    <input
      id="email"
      onChange={[Function]}
      value="an email"
    />
    <br />
    <h3>
      Business Address (optional)
    </h3>
    <br />
    Street
    <input
      id="street"
      onChange={[Function]}
      value="a street"
    />
    <br />
    City
    <input
      id="city"
      onChange={[Function]}
      value="a city"
    />
    <br />
    State
    <input
      id="state"
      onChange={[Function]}
      value="a state"
    />
    <br />
    Zip
    <input
      id="zip"
      onChange={[Function]}
      value="a zip"
    />
    <br />
    <h2>
      Payment Methods
    </h2>
    <input
      id="paymentMethods"
      onChange={[Function]}
      value="some payment methods"
    />
    <br />
    <h2>
      Social Media
    </h2>
    Facebook
    <input
      id="facebook"
      onChange={[Function]}
      value="a facebook link"
    />
    <br />
    Instagram
    <input
      id="instagram"
      onChange={[Function]}
      value="an instagram link"
    />
    <br />
    Twitter
    <input
      id="twitter"
      onChange={[Function]}
      value="a twitter link"
    />
    <br />
    <button>
      SAVE
    </button>
  </div>
</div>
`;
